File: Holiday/array_kata.py
myArray = [2,3,59,9,4]

def maximum_in(myArray = []):
    maximum = myArray[0]
    for count in range(0,len(myArray)):
        if myArray[count] > maximum:
            maximum = myArray[count]
    return maximum

print(maximum_in(myArray))


def minimum_in(myArray = []):
    minimum = myArray[0]
    for count in range(0,len(myArray)):
        if myArray[count] < minimum:
            minimum = myArray[count]
    return minimum

print(minimum_in(myArray))


def sum_of_array(myArray = []):
    sum = 0
    for count in range(0,len(myArray)):
        sum += myArray[count]
    return sum

print(sum_of_array(myArray))


def sum_of_even(myArray = []):
    sum = 0
    for count in range(0,len(myArray)):
        if myArray[count] % 2 == 0:
            sum += myArray[count]
    return sum

print(sum_of_even(myArray))


def sum_of_odd(myArray = []):
    sum = 0
    for count in range(0,len(myArray)):
        if myArray[count] % 2 != 0:
            sum += myArray[count]
    return sum

print(sum_of_odd(myArray))


def maximum_minimum(myArray = []):
    maximum = myArray[0]
    minimum = myArray[0]
    newArray = []
    for count in range(0,len(myArray)):
        if myArray[count] > maximum:
            maximum = myArray[count]
        if myArray[count] < minimum:
            maximum = myArray[count]
    newArray.append(maximum)
    newArray.append(minimum)  
    return newArray

print(maximum_minimum(myArray))


def number_of_odd(myArray = []):
    counter = 0
    for count in range(0,len(myArray)):
        if myArray[count] % 2 != 0:
            counter +=1
    return counter

print(number_of_odd(myArray))


def number_of_even(myArray = []):
    counter = 0
    for count in range(0,len(myArray)):
        if myArray[count] % 2 == 0:
            counter +=1
    return counter

print(number_of_even(myArray))

def array_of_even(myArray = []):
    newArray = []
    for count in range(0,len(myArray)):
        if myArray[count] % 2 == 0:
            newArray.append(myArray[count])
    return newArray

print(array_of_even(myArray))

def array_of_odd(myArray = []):
    newArray = []
    for count in range(0,len(myArray)):
        if myArray[count] % 2 != 0:
            newArray.append(myArray[count])
    return newArray

print(array_of_odd(myArray))


def square_of_array(myArray = []):
    newArray = []
    for count in range(0,len(myArray)):
        square = myArray[count] * 2
        newArray.append(square)
    return newArray

print(square_of_array(myArray))

File: Holiday/ArrayKata.java
import java.util.Arrays;

public class ArrayKata{
public static void main(String[] args){

int[] myArray = {4,5,8,9,3,2,7,6};

System.out.println(maximumIn(myArray));
System.out.println(minimumIn(myArray));
System.out.println(sumOfArray(myArray));
System.out.println(sumOfEven(myArray));
System.out.println(sumOfOdd(myArray));
System.out.println(Arrays.toString(maximumAndMinimum(myArray)));
System.out.println(numberOfEven(myArray));
System.out.println(numberOfOdd(myArray));
System.out.println(Arrays.toString(arrayOfOdd(myArray)));
System.out.println(Arrays.toString(arrayOfEven(myArray)));
System.out.println(Arrays.toString(squareArray(myArray)));
}

public static int maximumIn(int[] myArray){

int maximum = myArray[0];

for(int counter = 0;counter < myArray.length;counter++){
if(myArray[counter] > maximum)

maximum = myArray[counter];

}

return maximum;
}


public static int minimumIn(int[] myArray){
 
int minimum = myArray[0];

for(int counter = 0;counter < myArray.length;counter++){
if(myArray[counter] < minimum)

minimum = myArray[counter];

}

return minimum;

}



public static int sumOfArray(int[] myArray){

int sum = 0;                                                    

for(int counter = 0;counter < myArray.length;counter++){
sum += myArray[counter];
}

return sum;

}



public static int sumOfEven(int[] myArray){

int sum = 0;                                                    

for(int counter = 0;counter < myArray.length;counter++){

if(myArray[counter] % 2 == 0){
sum += myArray[counter];
}

}

return sum;

}



public static int sumOfOdd(int[] myArray){

int sum = 0;
                                                    

for(int counter = 0;counter < myArray.length;counter++){

if(myArray[counter] % 2 != 0){
sum += myArray[counter];
}

}

return sum;

}



public static int[] maximumAndMinimum(int[] myArray){

int number = myArray[0];


int[] newArray = new int[2];

for(int counter = 0;counter < myArray.length;counter++){
if(myArray[counter] < number)

newArray[0] = myArray[counter];

}
for(int counter = 0;counter < myArray.length;counter++){
if(myArray[counter] > number)

newArray[1] = myArray[counter];

}

return newArray;

}



public static int numberOfEven(int[] myArray){

int count = 0;                                                    

for(int counter = 0;counter < myArray.length;counter++){

if(myArray[counter] % 2 == 0){
count += 1;
}

}

return count;

}


public static int numberOfOdd(int[] myArray){

int count = 0;                                                    

for(int counter = 0;counter < myArray.length;counter++){

if(myArray[counter] % 2 != 0){
count += 1;
}

}

return count;

}


public static int[] arrayOfOdd(int[] myArray){

int index = 0;

int[] newArray = new int[(myArray.length / 2)];                                                    

for(int counter = 0;counter < myArray.length;counter++){

if(myArray[counter] % 2 != 0){
newArray[index] = myArray[counter];
index++;
}

}

return newArray;

}

public static int[] arrayOfEven(int[] myArray){

int index = 0;

int[] newArray = new int[(myArray.length / 2)];                                                    

for(int counter = 0;counter < myArray.length;counter++){

if(myArray[counter] % 2 == 0){
newArray[index] = myArray[counter];
index++;
}

}

return newArray;

}

public static int[] squareArray(int[] myArray){

int[] newArray = new int[(myArray.length)];                                                    

for(int counter = 0;counter < myArray.length;counter++){

newArray[counter] = myArray[counter] * 2;

}

return newArray;

}

}
